transports/http/endpoints/pulse_endpoints.go

```diff
@@ -34,6 +35,10 @@ func SetupPulseRoutes(s *service.Services, cfg *config.HTTPConfig) httpserver.Gi
 		merkleroots.NewHandler(s),
 	}
 
+	if cfg.EnablePprof {
+		routes = append(routes, profile.NewHandler(s))
+	}
+
```

Comment:
I would consider moving this if to handler. So instead you could have:
```suggestion
		merkleroots.NewHandler(s),
		profile.NewHandler(s, cfg),
	}
```